layouts/partials/footer.ejs

```diff
@@ -65,6 +66,19 @@
     $('.btn-user-initial').text(initials.join(''));
   }
 
+  // ajax for latest filenames, req will 304 and serve cache if unchanged
+  $.ajax({
+    method: 'GET',
+    url: '/filename-listing.json',
+    json: true
+  }).always(function(data, _, xhr) {
+    localStorage.setItem('filenames', JSON.stringify(data))
+  })
+
+  function getFilenameStorage() {
+    return JSON.parse(localStorage.getItem('filenames')).html
```

Comment:
Oh, meant to mention this as well...
Currently the `cache.set` method places the content passed to it in a key called `html`. Since this is not HTML being placed in the cache, I considered changing the key to something more generic like `content`, but doing so would mean previously cached items would become invalid when users upgrade their library to include this PR.

It would be fine for those using the default in-memory cache, but for those using redis stores/other caches that are not purged on app boot, I could see a change like this creating headaches.

Reply:
I think we could avoid this by falling back on the tree of files that is currently stored in memory. 

The main reason I can think of to serve the files list the way you are doing here is to avoid a JSON.stringify() call when the names come back. But I like your suggestion of checking a timestamp on the local storage data and keeping that limit relatively short (10 minutes, perhaps). That would keep the file listing relatively up to date but remove most of the extra requests on page navigation, which ought to help a lot with the load.

Taking that approach, we wouldn't need to interface with the caching layer at all and instead could move the endpoint logic into the `pages` route. We'd need to update the `updateTree()` function in `list.js` to return the file names in drive in addition to the drive tree, but that seems like a pretty easy change. 

What do you think?